CONTRIBUTING.md

```diff
@@ -1,16 +1,239 @@
+filling out the fields.
+
+Each bug report issue uses a template with 5 sections that are here to help
+maintainer understand your issue and eventually reproduce it:
+
+    #### Description
+
+    #### Steps to reproduce the issue
+
+    #### Expected results
+
+    #### Actual results
+
+    #### Versions
+
+To fill the `version` section, use the script provided in the RIOT git
+repository:
+```
+./dist/tools/ci/print_toolchain_versions.sh
+```
+
+In summary, try to put as much information as possible that you think could
+help maintainers or other developers to fix the bug quickly.
+
+## Pull Requests
+[pull-requests]: #pull-requests
+
+Pull Requests (PR) are the primary mechanism used to change RIOT. GitHub itself
+has some great documentation on
+[using the Pull Request feature][about-pull-requests].
+We use the [fork and pull model][development-models], where contributors push
+changes to their personal fork and create pull requests to bring those changes
+into the source repository.
+
+[about-pull-requests]: https://help.github.com/articles/about-pull-requests/
+[development-models]: https://help.github.com/articles/creating-a-pull-request-from-a-fork
+
+* [General rules](#general-rules)
+* [Git usage](#git-usage)
+
+### General rules
+[general-rules]: #general-rules
+
+* Before opening a new Pull Request, have a look at
+  [existing ones](https://github.com/RIOT-OS/RIOT/pulls). Maybe someone has
+  already opened one about the same thing and if it's the case, you might be
+  able to help with the contribution. Just comment on the PR and ask.
+
+* Each Pull Request form uses a template with 3 sections that are here to help
+  maintainer understand your contribution and help them testing it:
+
+      #### Contribution description
+
+      #### Testing procedure
+
+      #### Issues/PRs references
+
+  Please fill each section with as much information as possible.
+
+* Remember that smaller PRs tend to be merged faster, so keep your changes as
+  atomic as possible: don't hesitate to split your PRs into smaller ones.
+
+* In the Pull Request form, we recommend that you leave the
+  "Allow edits from maintainers" check box ticked. This will allow maintainer
+  finalizing your PR by pushing in your branch. In general, this speeds up the
+  PR merge in the main repository. Note that this is not an obligation.
+
+* Check if your code follows the
+  [coding conventions](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions).
+  For example, you can [uncrustify](http://uncrustify.sourceforge.net) a file:
+
+      $ uncrustify -c $RIOTBASE/uncrustify-riot.cfg <your file>
+
+* RIOT provides static test tools to verify the quality of changes (flake8,
+  trailing whitespaces, documentation, etc). These tools are wrapped in a
+  single `make` target: `static-test`
+
+      $ make static-test
+
+  Use it before opening a PR to perform last time checks.
+  *watch out:* this command will rebase your branch on your master branch.
```

Comment:
Maybe move it above the command instructions, so people reading line-by-line don't stumble over this *after* they executed the command ;-).